src/maintainer/knowledge_base.rst

```diff
@@ -1028,9 +1028,126 @@ In order to qualify as a noarch python package, all of the following criteria mu
   which builds on Linux `and` Windows, with ``build_number`` offsets to create a pair packages, like
   ``dataclasses``.
 
+.. hint::
+  
+  You can build platform-specific ``noarch`` packages to include runtime requirements depending on the target OS.
+  See mini-tutorial below.
+
 If an existing python package qualifies to be converted to a noarch package, you can request the required changes
 by opening a new issue and including ``@conda-forge-admin, please add noarch: python``.
 
+.. _os_specific_noarch:
+
+Noarch packages with OS-specific dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is possible to build ``noarch`` packages with runtime requirements that depend on the target OS
+(Linux, Windows, MacOS), regardless the architecture (amd64, ARM, PowerPC, etc). This approach
+relies on three concepts:
+
+1.  `Virtual packages <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html>`__. 
+    Prefixed with a double underscore, they are used by conda to represent system properties as
+    constraints for the solver at install-time. We will use ``__linux``, ``__win`` or ``__osx``,
+    which are only present when the running platform is Linux, Windows, or MacOS, respectively.
+    ``__unix`` is present in both Linux and MacOS. Note that this feature is **only fully available
+    on conda 4.10 or above**.
+2.  ``conda-forge.ymls``'s :ref:`noarch_platforms` option.
+
+The idea is to generate different noarch packages for each OS needing different dependencies.
+Let's say you have a pure Python package, perfectly eligible for ``noarch: python``, but on Windows
+it requires ``windows-only-dependency``. You might have something like:
+
+.. code-block:: yaml
+  :caption: recipe/meta.yaml (original)
+
+  name: package
+  source:
+    # ...
+  build:
+    number: 0
+  requirements:
+    # ...
+    run:
+      - python
+      - numpy
+      - windows-only-dependency  # [win]
+
+Being non-noarch, this means that the build matrix will include at least 12 outputs: three platforms,
+times four Python versions. This gets worse with arm64, aarch64 and ppc64le in the mix. We can get it down
+to two outputs if replace it with this other approach!
+
+.. code-block:: yaml+jinja
+  :caption: recipe/meta.yaml (modified)
+
+  name: package
+  source:
+    # ...
+  build:
+    number: 0
+    # You NEED to include the platform in the build string to avoid hash collisions
+    string: "unix_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}"  # [unix]
+    string: "win_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}"   # [win]
```

Comment:
Would it be possible to avoid manipulating the `string` and instead rely on different hashes (note Jaime is mentioning something similar elsewhere)?

The main issue with `string` is pretty finicky. Users need to be very careful to replicate conda-build's behavior and if they missing something they could wind up with duplicate packages (which sometimes can both be accepted by Anaconda.org with neither being downloadable).

Reply:
This will be needed until https://github.com/conda/conda-build/pull/4606 lands and a new conda-build release is cut.

Is it ok to merge as is or would you rather have a little admonition mentioning this caveat?

Reply:
My understanding from the thread above is using `conda_build_config.yaml` meant we would not need this.

Fixing conda-build to not need either is certainly better. That said, between setting a `string` and using `conda_build_config.yaml`, would prefer the latter (fewer foot guns).

Reply:
I am partial to `conda_build_config.yaml` too as we transition to the new conda-build version.